lib/crypt.c

```diff
@@ -138,10 +150,17 @@ do_crypt (const char *phrase, const char *setting, struct crypt_data *data)
          and special markers in the passwd(5) and shadow(5) files.)"
 
      However, individual hash function implementations have not always
-     enforced this rule, so for now we are being cautious and
-     rejecting only ':' and '\n'.  See
-     <https://github.com/besser82/libxcrypt/issues/105>.  */
-  if (strcspn (setting, ":\n") != set_size)
+     enforced this rule, so for now we are being cautious and consider
+     the following characters to be valid for any salt string in this
+     generic check:
+
+         0 1 2 3 4 5 6 7 8 9
+         a b c d e f g h i j k l m n o p q r s t u v w x y z
+         A B C D E F G H I J K L M N O P Q R S T U V W X Y Z
+         . / + - _ = , $ " # % & ' ( ) < > ? @ [ ] ^ ` { | } ~
```

Comment:
FWIW, I didn't verify that this indeed the exact list of characters not included among the bad ones.

Reply:
This is verified by `test/special-char-salt.c`.